meta/data.go

```diff
@@ -34,6 +36,15 @@ type Data struct {
 	MaxNodeID       uint64
 	MaxShardGroupID uint64
 	MaxShardID      uint64
+
+	Logger *log.Logger
+}
+
+// NewData returns a new instance of Data.
+func NewData() *Data {
+	return &Data{
+		Logger: log.New(os.Stderr, "[meta] ", log.LstdFlags),
```

Comment:
This seems odd to embed a logger into each `Data` instance.  It seems like we could just have a global logger in the `meta` package and use that.

Something like:
```
var Logger = log.New(os.Stdout, "[meta] ", log.LstdFlags)

Logger.Printf("new shard ID %d ...", ...)
```